### PR TITLE
Certora fixes

### PR DESCRIPTION
**Pull request by 1kresh**

### CODE REVIEW

File: src/contracts/vault/Vault.sol [hunk 1]
```diff
@@ -121,7 +120,10 @@ contract Vault is VaultStorage, MigratableEntity, AccessControlUpgradeable, Reen
     /**
      * @inheritdoc IVault
      */
-    function withdraw(address claimer, uint256 amount) external returns (uint256 burnedShares, uint256 mintedShares) {
+    function withdraw(
```
Comment (algys) [inline src/contracts/vault/Vault.sol:123]: missing re-entrancy

File: src/contracts/vault/Vault.sol [hunk 2]
```diff
@@ -130,36 +132,47 @@ contract Vault is VaultStorage, MigratableEntity, AccessControlUpgradeable, Reen
...
-        uint256 withdrawals_ = withdrawals[epoch];
-        uint256 withdrawalsShares_ = withdrawalShares[epoch];
+        emit Withdraw(msg.sender, claimer, amount, burnedShares, mintedShares);
+    }
 
-        mintedShares = ERC4626Math.previewDeposit(amount, withdrawalsShares_, withdrawals_);
+    /**
+     * @inheritdoc IVault
+     */
+    function redeem(
```
Comment (algys) [inline src/contracts/vault/Vault.sol:149]: non-reentrant

**Review by algys — CHANGES_REQUESTED** [review]

File: src/contracts/vault/Vault.sol (hunk 1 above)
Comment (1kresh) [reply src/contracts/vault/Vault.sol:123]: Fixed at https://github.com/symbioticfi/core/pull/27/commits/d2b8ca7c2ddc95ccfbfe6e8af4bad0d94b4aa0ca

File: src/contracts/vault/Vault.sol (hunk 2 above)
Comment (1kresh) [reply src/contracts/vault/Vault.sol:149]: Fixed at https://github.com/symbioticfi/core/pull/27/commits/d2b8ca7c2ddc95ccfbfe6e8af4bad0d94b4aa0ca

**Review by algys — APPROVED** [review]

**Pull request merged**